components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2061,6 +2061,25 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return team;
     }
 
+    public async updateTeam(ctx: TraceContext, team: Partial<Team>): Promise<Team> {
+        const teamId = team.id;
+        traceAPIParams(ctx, { teamId });
+
+        if (!teamId || !uuidValidate(teamId)) {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, "team ID must be a valid UUID");
+        }
+        this.checkUser("updateTeam");
+        const existingTeam = await this.teamDB.findTeamById(teamId);
+        if (!existingTeam) {
+            throw new ResponseError(ErrorCodes.NOT_FOUND, `Team ${team.id} does not exist`);
+        }
+        const members = await this.teamDB.findMembersByTeam(teamId);
+        await this.guardAccess({ kind: "team", subject: existingTeam, members }, "update");
```

Comment:
For others reading this, the update action is only allowed if you're the team owner.